Source/MLXNN/Pooling.swift

```diff
@@ -0,0 +1,124 @@
+// Copyright © 2024 Apple Inc.
+
+import Foundation
+import MLX
+
+open class Pool: Module, UnaryLayer {
+
+    public let kernelSize: [Int]
+    public let stride: [Int]
+    public let axes: [Int]
+    public let poolingOp: (MLXArray, [Int]) -> MLXArray
+
+    init(
+        kernelSize: [Int],
+        stride: [Int],
+        poolingOp: @escaping (MLXArray, [Int]) -> MLXArray
+    ) {
+        self.kernelSize = kernelSize
+        self.stride = stride
+        self.axes = Array((-1 * kernelSize.count - 1) ..< -1)
+        print(axes)
+        self.poolingOp = poolingOp
+    }
+
+    public func callAsFunction(_ x: MLX.MLXArray) -> MLX.MLXArray {
```

Comment:
I just discovered that even in an open class you cannot override public functions in a subclass.  Can you please change this to `open func callAsFunction()`?

I am prepping a PR for the rest of the NN classes.  Thanks!

Reply:
Done!